macros/PGmatrixmacros.pl

```diff
@@ -768,10 +768,12 @@ =head2 convert_to_array_ref {
 
 sub convert_to_array_ref {
 	my $input = shift;
-	if (Value::classMatch($input,"Matrix")) {
+	if (Value::isValue($input) && Value::classMatch($input,"Matrix")) {
```

Comment:
I would revert this file back to its original form.  No need to `Value::isValue($input)` with the correction above.  Also the `warn` lines below need to be removed.  So no actual changes to this file.

Reply:
I agree on removing the warn lines.  

However removing Value::isValue($input)
makes the test case fail (see expanded test case below). 

I still think this glitch should be fixed in Value::classMatch or perhaps Value::class() so that Value::classMatch
returns true only if $input is of type Value.  even if there are other objects as in this case that have the same name e.g. "Matrix".  Was Value::class supposed to return true for anything except Value types?

Reply:
I understand now that the subclassing refers to the classMatch method not to value.